src/System.Text.Json/System/Text/Json/JsonWriter.cs

```diff
@@ -6,193 +6,592 @@
 
 namespace System.Text.Json
 {
-    public struct JsonWriter<TFormatter> where TFormatter: ITextOutput
+    public struct JsonWriter
```

Comment:
I was used to avoid boxing implementations of ITextOutut and to devirtualize calls to members of the interface.